Vsphere permission update

dharapvj opened the pull request:
Updated documentation for vsphere permissions across kubeone and kkp for external CCM and CSI to work as experienced during migration in customer location.

CC: @toschneck

toschneck commented:
@dharapvj  a view points for the review

* we would need to change also the KKP docu (master, 2.18, 2.19) https://docs.kubermatic.com/kubermatic/master/architecture/support_policy/provider_support_matrix/vsphere/vsphere/
* we should add a cross reference link to the vsphere CSI driver page
* we need a hint that the permissions needs to get updated with 2.18 (docu page itself and potential in the changelog / upgrade instructions)

toschneck commented:
/assign @dharapvj

dharapvj commented:
@toschneck 


>   we would need to change also the KKP docu (master, 2.18, 2.19) https://docs.kubermatic.com/kubermatic/master/architecture/support_policy/provider_support_matrix/vsphere/vsphere/

Those are the exact pages, I had updated. So we are good there I suppose.

>    we should add a cross reference link to the vsphere CSI driver page

Added.

>    we need a hint that the permissions needs to get updated with 2.18 (docu page itself and potential in the changelog / upgrade instructions)

Not sure how to update changelog for 2.18. Can you guide?

dakraus commented on content/kubermatic/master/architecture/support_policy/provider_support_matrix/vSphere/vSphere.en.md:
```diff
@@ -225,4 +229,5 @@ are not supported yet at `Cluster` level by Kubermatic UI.
 
-[vsphere-cloud-config]: https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html#template-config-file
+[vsphere-cloud-config]: https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-BFF39F1D-F70A-4360-ABC9-85BDAFBE8864.html?hWord=N4IghgNiBcIMYQK4GcAuBTATgWgJYBMACAYQGUBJEAXyA
+[vsphere-storage-plugin-roles]: https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-043ACF65-9E0B-475C-A507-BBBE2579AA58.html
```
@dharapvj the link is unfortunately no longer valid.... Maybe it's now this guide?!
* https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-0AB6E692-AA47-4B6A-8CEA-38B754E16567.html#GUID-043ACF65-9E0B-475C-A507-BBBE2579AA58__GUID-E51466CB-F1EA-4AD7-A541-F22CDC6DE881

dharapvj replied:
Done in latest commit

toschneck reviewed (changes requested):
Missing things:
- [ ] update link
- [ ] update `2.20` content
- [ ] if possible, we should potential execute `govc role.ls` and add the output, like this:
  ```
  govc role.ls k8c-storage-datastore-vsan
  Host.Config.Storage
  System.Anonymous
  System.Read
  System.View
  ```

dharapvj commented:
All review comments addressed from @dakraus and @toschneck .. please check again

toschneck commented:
/lgtm

toschneck commented:
/approve
